sklearn/utils/validation.py

```diff
@@ -155,6 +156,36 @@ def _shape_repr(shape):
     return "(%s)" % joined
 
 
+def check_memory(memory):
+    """Check that ``memory`` is joblib.Memory-like.
+
+    joblib.Memory-like means that ``memory`` can be converted into a
+    sklearn.externals.joblib.Memory instance (typically a str denoting the
+    ``cachedir``) or has the same interface.
+
+    Parameters
+    ----------
+    memory : joblib.Memory-like
+
+    Returns
+    -------
+    memory : object with the joblib.Memory interface
+
+    Raises
+    ------
+    ValueError
+        if ``memory`` is not joblib.Memory-like
+    """
+
+    if memory is None or isinstance(memory, six.string_types):
+        memory = Memory(cachedir=memory, verbose=0)
+    elif not hasattr(memory, 'cache'):
+        raise ValueError("'memory' should be None, a string or have the same"
+                         " interface as sklearn.externals.joblib.Memory."
+                         " Got {} instead.".format(memory))
```

Comment:
I would use "Got memory='{}' instead." for better readability. You will need to update the tests accordingly.

Reply:
Done.